torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -122,7 +127,7 @@ class TORCH_API RpcAgent {
   const WorkerInfo workerInfo_;
   const std::string workerName_;
   const std::unique_ptr<RequestCallback> cb_;
-  const std::chrono::milliseconds rpcTimeout_;
+  std::chrono::milliseconds rpcTimeout_;
```

Comment:
Doesn't this need to be atomic since `setRpcTimeout` could be called from a different thread while the watchdog thread is reading from it?

Reply:
@pritamdamania87 The set function would only be invoked via pybind from a python call, and the GIL is not dropped, so should we be fine for thread safety? I don't anticipate needing to call `setRPCTimeout` from C++.

Reply:
it might just be worth grabbing the lock in setRpcTimeout() ... compared with all the other things going on, this should be super cheap and help us avoid some unexpected corner-cases.

Reply:
@rohan-varma We probably hold the GIL for writing (setRpcTimeout), but no lock is held while reading.